core/input/input_event.cpp

```diff
@@ -731,12 +738,23 @@ bool InputEventJoypadMotion::action_match(const Ref<InputEvent> &p_event, bool *
 				if (p_deadzone == 1.0f) {
 					*p_strength = 1.0f;
 				} else {
-					*p_strength = CLAMP(Math::inverse_lerp(p_deadzone, 1.0f, Math::abs(jm->get_axis_value())), 0.0f, 1.0f);
+					*p_strength = CLAMP(Math::inverse_lerp(p_deadzone, 1.0f, jm_abs_axis_value), 0.0f, 1.0f);
 				}
 			} else {
 				*p_strength = 0.0f;
 			}
 		}
+		if (p_raw_strength != nullptr) {
```

Comment:
There is something really weird going on here. The `raw_strength` is not supposed to depend on the deadzone at all.
The raw strength should basically be something like:
```c++
if (same_direction) {
    *p_raw_strength = CLAMP(jm_abs_axis_value, 0.0f, 1.0f);
} else {
    *p_raw_strength = 0.0f;
}
```
or in one line:
```c++
*p_raw_strength = same_direction ? CLAMP(jm_abs_axis_value, 0.0f, 1.0f) : 0.0f;
```

Reply:
I'm surprised that I missed this, but I don't think the clamping is necessary. I think it's guaranteed to be between 0 and 1 since it's the absolute value of a value that [says it's between -1 and 1](https://github.com/godotengine/godot/blob/master/core/input/input_event.h#L388). I think this also applies to `p_strength` but I didn't want to change too much to avoid breaking things.